src/js/components/Search.jsx

```diff
@@ -1,33 +1,166 @@
+import SourceSelect from './SourceSelect'
+import {DataTable} from 'safe-framework'
+import {RaisedButton} from 'material-ui'
+
+import {
+  addFilter,
+  editFilter,
+  fetchSearchResults,
+  fetchSourceFields,
+  fetchSources,
+  setSource
+} from '../actions'
+
 import {header, main} from '../styles/common'
 
+const style = {
+  hidden: {
+    display: 'none'
+  },
+  verticalTop: {
+    verticalAlign: 'top'
+  }
+}
+
 class Search extends Component {
+  static propTypes = {
+    dispatch: PropTypes.func.isRequired,
+    filters: PropTypes.array.isRequired,
+    searchResults: PropTypes.object.isRequired,
+    source: PropTypes.string.isRequired
+  }
+
+  constructor (props) {
+    super(props)
+
+    this.onAddFilter = ::this.onAddFilter
+    this.onChangeField = ::this.onChangeField
+    this.onChangeOperator = ::this.onChangeOperator
+    this.onChangeSource = ::this.onChangeSource
+    this.onChangeValue = ::this.onChangeValue
+    this.onClick = ::this.onClick
+  }
+  
+  componentWillMount () {
+    const {dispatch} = this.props
+
+    dispatch(fetchSources())
+  }
+
+  onAddFilter (ev) {
+    const {dispatch} = this.props
+
+    ev.preventDefault()
+    dispatch(addFilter({
+      field: '',
+      operator: '',
+      value: ''
+    }))
+  }
+  
+  onChangeField (index, field) {
+    const {dispatch} = this.props
+
+    dispatch(editFilter(index, {field}))
+  }
+
+  onChangeOperator (index, operator) {
+    const {dispatch} = this.props
+
+    dispatch(editFilter(index, {operator}))
+  }
+
+  onChangeSource (ev, index, source) {
+    const {dispatch} = this.props
+
+    ev.preventDefault()
+    dispatch(setSource(source))
+    dispatch(fetchSourceFields(source))
+  }
+
+  onChangeValue (index, value) {
+    const {dispatch} = this.props
+
+    dispatch(editFilter(index, {value}))
+  }
+  
+  onClick () {
+    const {dispatch, source, filters} = this.props
+    
+    dispatch(fetchSearchResults(source, filters))
+  }
+  
   render () {
+    const {source, searchResults} = this.props
+    const filterStyle = source === ''
+      ? style.hidden
+      : {}
+      
     return (
       <div>
         <header style={header}>
           <h1>Search</h1>
         </header>
         <main style={main}>
-          <h3>Select a data source</h3>
-          <h3>Select filter criteria (optional)</h3>
-          <TextField
-            floatingLabelText='Filter Criteria'
-            hintText='Filter Criteria'
+          <SourceSelect
+            style={style.verticalTop}
+            onChange={this.onChangeSource}
+          />
+          <FilterCriteria
+            style={{
+              ...style.verticalTop
+            }}
+            wrapperStyle={{
+              ...filterStyle
+            }}
+            onAdd={this.onAddFilter}
+            onChangeField={this.onChangeField}
+            onChangeOperator={this.onChangeOperator}
+            onChangeValue={this.onChangeValue}
+          />
+          <br/>
```

Comment:
Instead of `<br />`, you sould add margins to each element. `<br />` tags can appear differently in each browser.